src/main/java/bl4ckscor3/mod/scserverutils/mute/MuteEventsHandler.java

```diff
@@ -0,0 +1,152 @@
+                        Component messageComponent = parseComponent(sender.level(), muteMessages.cancelledMessageSuspend().get());
+                        sender.sendSystemMessage(messageComponent);
+                    }
+                } catch (CommandSyntaxException ignored) {}
+
+            } else if (cmd.equalsIgnoreCase("teammsg") || cmd.equalsIgnoreCase(("say"))) {
+                try {
+                    ServerPlayer sender = event.getParseResults().getContext().getSource().getPlayerOrException();
+                    if (mutedPlayersUUID.contains(sender.getStringUUID())) {
+                        logCancelledMessage(sender, String.join(" ", Arrays.copyOfRange(args, 1, args.length)));
+                        event.setCanceled(true);
+                        MuteMessages muteMessages = Configuration.instance.muteMessages;
+                        Component messageComponent = parseComponent(sender.level(), muteMessages.cancelledMessageMute().get());
+                        sender.sendSystemMessage(messageComponent);
+                    } else if (sender.getTags().contains("suspended")) {
+                        logCancelledMessage(sender, String.join(" ", Arrays.copyOfRange(args, 1, args.length)));
+                        event.setCanceled(true);
+                        MuteMessages muteMessages = Configuration.instance.muteMessages;
+                        Component messageComponent = parseComponent(sender.level(), muteMessages.cancelledMessageSuspend().get());
+                        sender.sendSystemMessage(messageComponent);
+                    }
+                } catch (CommandSyntaxException ignored) {}
+            }
+        }
+    }
+
+    @SubscribeEvent
+    public static void onServerStarted(ServerStartedEvent event) {
+        Path serverPath = event.getServer().getServerDirectory();
+        playerDataManager = new PlayerDataManager(serverPath);
+        for (PlayerMuteData playerData: playerDataManager.getEntries()) {
+            mutedPlayersUUID.add(playerData.uuid);
+        }
+
+    }
+
+    @SubscribeEvent
+    public static void onServerStopping(ServerStoppingEvent event) {
+        if (playerDataManager != null) {
+            playerDataManager.save();
+        }
+    }
+
+    @SubscribeEvent
+    public static void onWorldSave(LevelEvent.Save event) {
+        if (event.getLevel() instanceof ServerLevel && ((ServerLevel) event.getLevel()).dimension() == Level.OVERWORLD) {
+            if (playerDataManager != null) {
+                playerDataManager.save();
+            }
+        }
+    }
+
+    @SubscribeEvent
+    public static void onBlockPlace(BlockEvent.EntityPlaceEvent event) {
+        if (event.getEntity() instanceof ServerPlayer) {
+            ServerPlayer p = (ServerPlayer) event.getEntity();
+
+            if ((event.getPlacedBlock().getBlock() instanceof SignBlock) || (event.getPlacedBlock().getBlock() instanceof WallSignBlock))
```

Comment:
Perhaps being able to configure which blocks are not placable during a mute/suspend period would be useful. This will make it easier to just add onto the blocks through the config if you find that you need any others, without requiring a mod update.